diff -rupN mali-bifrost.orig/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c mali-bifrost/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c
--- mali-bifrost.orig/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c	2024-06-13 14:14:18.000000000 -0400
+++ mali-bifrost/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c	2026-08-28 21:24:08.044543826 -0400
@@ -4557,7 +4557,7 @@ regulators_probe_defer:
 	}
 #endif
 
-clocks_probe_defer:
+// clocks_probe_defer:
 #if defined(CONFIG_REGULATOR)
 	for (i = 0; i < BASE_MAX_NR_CLOCKS_REGULATORS; i++)
 		regulator_put(kbdev->regulators[i]);
