diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 111111111111..222222222222 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -140,6 +140,8 @@ struct panel_simple {
 	struct gpio_desc *spi_scl_gpio;
 	struct gpio_desc *spi_cs_gpio;
 	struct device_node *np_crtc;
+
+	enum drm_panel_orientation orientation;
 };
 
 enum rockchip_cmd_type {
@@ -727,6 +729,8 @@ static int panel_simple_get_modes(struct drm_panel *panel,
     /* Then timing / fixed modes from descriptor */
     num += panel_simple_get_non_edid_modes(p, connector);
 
+    drm_connector_set_panel_orientation(connector, p->orientation);
+
     return num;
 }
 
@@ -874,6 +878,13 @@ static int panel_simple_probe(struct device *dev, const struct panel_desc *desc)
 		}
 	}
 
+	err = of_drm_get_panel_orientation(dev->of_node, &panel->orientation);
+	if (err) {
+		dev_err(dev, "%pOF: failed to get orientation %d\n",
+			dev->of_node, err);
+		return err;
+	}
+
 	/* RG353V is a DSI panel, so use DSI connector type */
 drm_panel_init(&panel->base, dev, &panel_simple_funcs,
                DRM_MODE_CONNECTOR_DSI);
