diff --git a/rocknix-singleadc-joypad.c b/rocknix-singleadc-joypad.c
--- a/rocknix-singleadc-joypad.c
+++ b/rocknix-singleadc-joypad.c
@@ -1811,6 +1813,9 @@
 /*----------------------------------------------------------------------------*/
 MODULE_AUTHOR("ROCKNIX");
 MODULE_DESCRIPTION("ROCKNIX singleadc joypad driver");
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0) && LINUX_VERSION_CODE < KERNEL_VERSION(6, 13, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
 MODULE_LICENSE("GPL");
 MODULE_ALIAS("platform:" DRV_NAME);
 MODULE_INFO(intree, "Y");
